beacon-chain/powchain/service.go

```diff
@@ -281,8 +286,8 @@ func (s *Service) ChainStartEth1Data() *ethpb.Eth1Data {
 
 // PreGenesisState returns a state that contains
 // pre-chainstart deposits.
-func (s *Service) PreGenesisState() state.BeaconState {
-	return s.preGenesisState
+func (s *Service) PreGenesisState() (state.BeaconState, error) {
```

Comment:
why does this need to return an error.

Reply:
It implements the same interface as `mock_faulty_powchain`:
https://github.com/prysmaticlabs/prysm/blob/06c1ef83114791b976785295208742e0506f756b/beacon-chain/powchain/testing/mock_faulty_powchain.go#L78-L84

Reply:
We shouldn't be changing a production method just to satisfy our test mocks